File: api/server/src/main/java/su/plo/voice/api/server/audio/source/ServerPlayerSource.java
package su.plo.voice.api.server.audio.source;

import org.jetbrains.annotations.NotNull;
import su.plo.voice.api.server.player.VoicePlayer;
import su.plo.voice.proto.data.audio.source.PlayerSourceInfo;

public interface ServerPlayerSource extends ServerAudioSource<PlayerSourceInfo> {

    @NotNull VoicePlayer getPlayer();
}

File: api/server/src/main/kotlin/su/plo/voice/api/server/audio/source/ServerPlayerSource.kt
package su.plo.voice.api.server.audio.source

import su.plo.voice.api.server.player.VoiceServerPlayer
import su.plo.voice.proto.data.audio.source.PlayerSourceInfo

/**
 * Represents a player audio source.
 */
interface ServerPlayerSource : ServerPositionalSource<PlayerSourceInfo> {

    /**
     * Gets the player associated with this audio source.
     *
     * @return The player.
     */
    val player: VoiceServerPlayer
}
